src/CrudPanel.php

```diff
@@ -237,16 +239,91 @@ public function sort($items)
      *          App/Models/Address defined by a company() method on the user model and an address() method on the
      *          company model, the 'App/Models/Address' string will be returned.
      *
-     * @param $relationString String Relation string. A dot notation can be used to chain multiple relations.
+     * @param string $relationString Relation string. A dot notation can be used to chain multiple relations.
+     * @param int $length Optionally specify the number of relations to omit from the start of the relation string. If
+     *        the provided length is negative, then that many relations will be omitted from the end of the relation
+     *        string.
+     * @param Model $model Optionally specify a different model than the one in the crud object.
      *
-     * @return string relation model name
+     * @return string Relation model name.
      */
-    private function getRelationModel($relationString)
+    public function getRelationModel($relationString, $length = null, $model = null)
     {
-        $result = array_reduce(explode('.', $relationString), function ($obj, $method) {
+        $relationArray = explode('.', $relationString);
+
+        if (! isset($length)) {
+            $length = count($relationArray);
+        }
+
+        if (! isset($model)) {
+            $model = $this->model;
+        }
+
+        $result = array_reduce(array_splice($relationArray, 0, $length), function ($obj, $method) {
             return $obj->$method()->getRelated();
-        }, $this->model);
+        }, $model);
 
         return get_class($result);
     }
+
+    /**
+     * Get the given attribute from a model or models resulting from the specified relation string (eg: the list of streets from
+     * the many addresses of the company of a given user).
+     *
+     * @param Model $model Model (eg: user).
+     * @param string $relationString Model relation. Can be a string representing the name of a relation method in the given
+     *        Model or one from a different Model through multiple relations. A dot notation can be used to specify
+     *        multiple relations (eg: user.company.address).
+     * @param string $attribute The attribute from the relation model (eg: the street attribute from the address model).
+     *
+     * @return array An array containing a list of attributes from the resulting model.
+     */
+    public function getModelAttributeFromRelation($model, $relationString, $attribute)
+    {
+        $lastModels = $this->getRelationModelInstance($model, $relationString);
+        $attributes = [];
+        foreach ($lastModels as $model) {
+            if ($model->{$attribute}) {
+                $attributes[] = $model->{$attribute};
+            }
+        }
```

Comment:
Out of morbid interest, is it possible for `$lastModels` to be the empty array (i.e. `[]`)?

Reply:
Yes, nothing will be shown in that case.